kasa/smartdevice.py

```diff
@@ -411,11 +424,13 @@ def has_children(self) -> bool:
         return self.is_strip
 
     @property  # type: ignore
-    @requires_update
+    @requires_discovery_or_update
     def alias(self) -> str:
         """Return device name (alias)."""
-        sys_info = self._sys_info
-        return str(sys_info["alias"])
+        if self._sys_info:
+            return str(self._sys_info["alias"])
+        else:
+            return str(self._discovery_info["device_type"])  # type: ignore
```

Comment:
See my later comments wrt. branching on the property accesses.

In this specific case, perhaps we should use some placeholder for unavailable aliases? "Unavailable alias"? "Authentication needed"? "Unknown"? I'm not sure about the wording, but something more approachable than using device_type as a magical alias is probably what we want to expose to users.

Reply:
Fixed and done - I went with "Authentication required"